amounts.go

```diff
@@ -4,19 +4,60 @@ import (
 	"fmt"
 	"math/big"
 	"regexp"
+	"strconv"
 
 	"github.com/pkg/errors"
-	samount "github.com/stellar/go/amount"
+	"github.com/stellar/go/xdr"
 )
 
-// Alow "-1", "1.", ".1", "1.1".
-// But not "." or "".
-var decimalStrictRE = regexp.MustCompile(`^-?((\d+\.?\d*)|(\d*\.?\d+))$`)
+// One is the value of one whole unit of currency. Stellar uses 7 fixed digits
+// for fractional values, thus One is 10 million (10^7).
+const (
+	One = 10000000
```

Comment:
Let's call this something different. Maybe StroopsPerLumen, PartsInWhole, StellarCurrencyFactor, even VeryLargeNumber or TenToThe7.

Reply:
+1. i like `StroopsPerLumen` and `StellarCurrencyFactor`. i think the first is a little more descriptive.